docs/backup-restore-integration.md

```diff
@@ -0,0 +1,287 @@
+
+- ("", "PersistentVolumeClaim", "ns1", "pvc1")
+
+#### spec.accessCredentials[\*].sshPublicKey
+
+```yaml
+...
+spec:
+  accessCredentials:
+  - sshPublicKey:
+      source:
+        secret:
+          secretName: my-pub-key
+...
+```
+
+- ("", "Secret", "ns1", "my-pub-key")
+
+#### spec.accessCredentials[\*].userPassword
+
+```yaml
+...
+spec:
+  accessCredentials:
+  - userPassword:
+      source:
+        secret:
+          secretName: my-user-password
+...
+```
+
+- ("", "Secret", "ns1", "my-user-password")
+
+### VirtualMachineInstanceReplicaSet Object Graph
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstanceReplicaSet
+metadata:
+  name: vmirs1
+  namespace: ns1
+...
+```
+
+- ("kubevirt.io", "VirtualMachineInstanceReplicaSet", "ns1", "vmirs1")
+- ("kubevirt.io", "VirtualMachineInstance", "ns1", "vmirs1XXXX1") \*
+- ("kubevirt.io", "VirtualMachineInstance", "ns1", "vmirs1XXXX2") \*
+
+\* there are usually multiple VirtualMachineInstances corresponding to a VirtualMachineInstanceReplicaSet.  The backup process can look up the name of this pod by using the `kubevirt.io/vmReplicaSet=<name of VirtualMachineInstanceReplicaSet>` label selector.
+
+#### spec.template
+
+See [VirtualMachineInstance](#virtualmachineinstance-object-graph)
+
+## Backup Actions
+
+### Guest filesystem freeze/thaw hooks
+
+See [this guide](https://github.com/kubevirt/kubevirt/blob/main/docs/freeze.md) for how to execute the freeze/thaw hooks for each VirtualMachineInstance encountered in the object graph.
+
+## Restore Actions
+
+### VirtualMachine Restore
+
+If restoring to a different cluster, and mac address or bios serial number are explicitly set, you should make sure there will be no collisions.  These values are set in:
+
+```
+/spec/template/spec/domain/devices/interfaces/<index>/macAddress
+/spec/template/spec/domain/firmware/serial
+```
+
+### VirtualMachineInstance Restore
+
+If a VirtualMachineInstance is **owned by a VirtualMachine**, it should **not be restored**.  The KubeVirt controller will recreate the resource based on the VirtualMachine definition.  Otherwise, VirtualMachineInstance definitions may be backed up/restored with the same precautions as VirtualMachine for mac address/bios.
+
+```
+/spec/domain/devices/interfaces/<index>/macAddress
+/spec/domain/firmware/serial
+```
+
+### virt-launcher Pod Restore
+
+A Pod with “virt-launcher-” prefix that is ***owned by a VirtualMachineInstance** should not be restored.
+
+### DataVolume Restore
+
+DataVolumes in **Succeeded phase** (status.phase) should have the following **annotation added** at restore time.  Otherwise, the associated PersistentVolumeClaim may get corrupted.  DataVolumes in any phase other than Succeeded do not need to be annotated.
+
+```yaml
+cdi.kubevirt.io/storage.prePopulated: <datavolume name>
+```
+
+### PersistentVolumeClaim Restore
+
+PersistentVolumeClaims ***owned by DataVolumes*** must have the following ***annotation added*** at backup/restore time.
```

Comment:
isn't `owned by datavolumes` not accurate? if this annotation is added to a pvc inprogress and not only after population is completed then we will not do the population?

Reply:
problem is that looking at just the PVC, you cannot tell whether the DataVolume is succeeded or not.  So maybe DataVolume controller can ignore this annotation on the PVC if DV is not succeeded?

Reply:
I believe that we add to the datavolume the "prepopulated" annotation once it is succeeded, so seems like the "populatedfor" on the pvc might be redudant, but I think it is used for other cases, if I recall correctly namespace transfer? where the dv is not yet succeeded but the pvc is already populated? or something like that

Reply:
pretty sure we never add "prepopulated" to the dv.  That would break the "delete the pvc and watch it get recreated/populated flow."

"populatedFor" is used by namespace transfer though, which is probably not ideal

Reply:
From what I see we do add prepopulated once the dv is set to succeeded. which is weird since you are right with the fact the if this annotation is set we dont recreate the PVC.

Reply:
Ignore my last comment it was on my local branch with the changes before we decided not to change current behavior.